s3tests_boto3/functional/test_iam.py

```diff
@@ -1795,3 +1796,290 @@ def test_copy_object_allow_get_src_put_dest():
     eq(etag, response['ETag'])
     response = client.delete_user_policy(PolicyName='AllowGetPolicy', UserName=get_main_user_id())
     eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+
+@attr(resource='user-policy')
+@attr(method='s3 Actions')
+@attr(operation='Verify Allow and Deny Get Bucket Versioning API using IAM policy for self')
+@attr(assertion='succeeds')
+@attr('user-policy')
+@attr('test_of_iam')
+def test_allow_deny_get_bucket_versioning_iam_policy():
+    client = get_iam_client()
+    s3_client_iam = get_iam_s3client()
+    bucket = get_new_bucket(client=s3_client_iam, name="iam1buk1")
```

Comment:
Don't give bucket name, let it autogenerate

```suggestion
    bucket = get_new_bucket(client=s3_client_iam)
```